server/src/main/java/io/pravega/schemaregistry/server/rest/resources/GroupResourceImpl.java

```diff
@@ -0,0 +1,690 @@
+            String group = createGroupRequest.getGroupName();
+            return getRegistryService().createGroup(namespace, group, groupProperties)
+                                  .thenApply(createStatus -> {
+                                      if (!createStatus) {
+                                          log.info("group {} exists", group);
+                                          return Response.status(Status.CONFLICT).build();
+                                      }
+                                      log.info("group {} created", group);
+                                      return Response.status(Status.CREATED).build();
+                                  })
+                                  .exceptionally(exception -> {
+                                      log.warn("createGroup failed with exception: ", Exceptions.unwrap(exception));
+                                      return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+                                  });
+        }).thenApply(response -> {
+            asyncResponse.resume(response);
+            return response;
+        });
+    }
+
+    @Override
+    public void getGroupProperties(String group, String namespace,
+                                   AsyncResponse asyncResponse) {
+        String resource = Strings.isNullOrEmpty(namespace) ? getGroupResource(group) : 
+                getGroupResource(group, namespace);
+        withCompletion("getGroupProperties", READ, resource, asyncResponse,
+                () -> getRegistryService().getGroupProperties(namespace, group)
+                                     .thenApply(groupProperty -> {
+                                         log.info("Group {} property found are {}", group, groupProperty);
+                                         return Response.status(Status.OK).entity(ModelHelper.encode(groupProperty)).build();
+                                     })
+                                     .exceptionally(exception -> {
+                                         if (Exceptions.unwrap(exception) instanceof StoreExceptions.DataNotFoundException) {
+                                             log.warn("Group {} not found", group);
+                                             return Response.status(Status.NOT_FOUND).build();
+                                         }
+                                         log.warn("getGroupProperties for group {} failed with exception: ", group, Exceptions.unwrap(exception));
+                                         return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+                                     }))
+                .thenApply(response -> {
+                    asyncResponse.resume(response);
+                    return response;
+                });
+    }
+    
+    @Override
+    public void getGroupHistory(String group, String namespace, AsyncResponse asyncResponse) {
```

Comment:
I see a lot of commonality in the last few methods and this. Could we extract out the common aspects into a separate method to reduce duplication?

Reply:
could you please elaborate which lines are similar?
typically all methods call a corresponding backend implementation method in `schemaregistryservice`.

there are two APIs that call the `getGroupHistory` backend method but that is where the similarity ends because they handle the response differently.